crawl-ref/source/fight.cc

```diff
@@ -517,10 +519,11 @@ bool fight_melee(actor *attacker, actor *defender, bool *did_hit, bool simu)
         if (!attacker->alive())
             return false;
 
-        // Monster went away?
+        // Monster went away or become friendly?
         if (!defender->alive()
             || defender->pos() != pos
-            || defender->is_banished())
+            || defender->is_banished()
+            || was_hostile && mons_aligned(attacker, defender))
```

Comment:
Wondering if there should be a check for if the monster is confused, berserk etc.? If they're not in control of their senses it wouldn't make sense for them to be aware to stop

Reply:
That makes sense to me. I believe we only need to check for confused and frenzied as that's what friendly monsters attacking the player checks (but this code could be buggy). We don't need to check marionette as that changes the monster's attitude and I don't think we need to check for vexed as that doesn't call `fight_melee` (although maybe we should check it anyway in case it does in the future?).